src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/InGameObjectFinder.cs

```diff
@@ -306,109 +348,286 @@ private RecordedGameObjectState CreateStateForTransform(bool replay, int screenW
 
                 if (onCamera)
                 {
+                    var objectTransformId = t.GetInstanceID();
+
+                    RecordedGameObjectState resultObject = null;
+                    var usingOldObject = false;
+                    if (priorState != null)
+                    {
+                        for (var i = 0; i < psCount; i++)
+                        {
+                            var priorObject = priorState[i];
+                            if (priorObject.id == objectTransformId)
+                            {
+                                resultObject = priorObject;
+                                usingOldObject = true;
+                                break;
+                            }
+                        }
+                    }
+                    if (resultObject == null)
+                    {
+                        resultObject = new RecordedGameObjectState()
+                        {
+                            id = objectTransformId,
+                            transform = t,
+                            path = GetUniqueTransformPath(t).Path,
+                            tag = statefulGameObject.tag,
+                            layer = LayerMask.LayerToName(statefulGameObject.layer),
+                            scene = statefulGameObject.scene,
+                            behaviours = new List<BehaviourState>(),
+                            colliders = new List<ColliderRecordState>(),
+                            worldSpaceBounds = null,
+                            rigidbodies = new List<RigidbodyRecordState>()
+                        };
+                    }
+
                     // make sure the screen space bounds has a non-zero Z size around 0
                     // we track the true z offset separately for ease of mouse selection on replay
-                    var size = new Vector3(maxX - minX, maxY - minY, 0.1f);
-                    var center = new Vector3(minX + size.x / 2, minY + size.y / 2, 0);
-                    var screenSpaceBounds = new Bounds(center, size);
+                    var extents = new Vector3((maxX - minX)/2, (maxY - minY)/2, 0.05f);
+                    var center = new Vector3(minX + extents.x, minY + extents.y, 0);
+
+                    var worldExtents = new Vector3((maxWorldX - minWorldX)/2, (maxWorldY - minWorldY)/2, (maxWorldZ - minWorldZ)/2);
+                    var worldCenter = new Vector3(minWorldX + worldExtents.x, minWorldY + worldExtents.y, minWorldZ + worldExtents.z);
```

Comment:
Are these `Vector3`s worth pre-allocating + reusing?

Reply:
could I, I guess technically yes, but not in a reasonable way

The way the unity code is setup for Bounds/Extents/etc you get a copy of the Vector3 instead of being able to reference the value.  

Take for example this sample code and note how changing the value of x on the extents does not affect in internal bounds extents object.  So while yes we 'can' do it, we unfortunately don't actually save any new object creations.

![image](https://github.com/Regression-Games/RGUnityBots/assets/112959031/b67db98a-380f-4b12-80e5-3b9125b03c60)

Reply:
makes sense. Thanks for clarifying